### PR TITLE
Analytics Performance: add pull-to-refresh tracks to analytics hub

### LINKED ISSUE #9441: [Analytics Performance] Introduce Analytics Hub Pull-to-refresh track event

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Fix  by creating a track event that will register every time a pull-to-refresh is triggered inside the Analytics Hub. This is necessary since PR #9363 will introduce a force refresh action every time the pull-to-refresh happens, overriding our cache system and skipping the performance strategy we're introducing.

How to Test
==========
1. Open the app and enter the Analytics Hub
2. Once the data is loaded, exit and reenter the Hub with the same range selection, and verify that no network operation is triggered.
3. Trigger a pull-to-refresh action. Now verify that a new request was made fetching the stats data inside the Hub again for that specific selection range, and the new track event is triggered as expected.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt [hunk 1]
```diff
@@ -170,6 +170,7 @@ enum class AnalyticsEvent(val siteless: Boolean = false) {
     // -- Analytics Hub
     ANALYTICS_HUB_DATE_RANGE_BUTTON_TAPPED,
     ANALYTICS_HUB_DATE_RANGE_SELECTED,
+    ANALYTICS_HUB_PULL_TO_REFRESH_TRIGGERED,
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt:173]: np: Is this event new? If yes, then I think we should register it and add the registration PR link to this PR. WDYT?

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/analytics/AnalyticsHubViewModelTest.kt [hunk 2]
```diff
@@ -644,6 +644,16 @@ class AnalyticsHubViewModelTest : BaseUnitTest() {
         verify(tracker).track(AnalyticsEvent.ANALYTICS_HUB_DATE_RANGE_BUTTON_TAPPED)
     }
 
+    @Test
+    fun `when onRefreshRequested is called, then trigger expected analytics event`() = testBlocking {
```
Comment (atorresveiga) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/analytics/AnalyticsHubViewModelTest.kt:648]: Nice!!!

**Review by atorresveiga — APPROVED** [review]
The code looks great and is working as expected! Excellent work adding the unit test. I left only a tiny comment related to the event registration.

Evidence:
Screenshot 2023-07-21 at 12 58 31

⚠️  Warning: This code has the issue of the selection being broken

LGTM! :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt:173]: Fair point! Let me fix this!

**Comment by ThomazFB** [comment]
Thank you, @atorresveiga! I'm addressing the event registration right now! I'll be back in a moment 😄 

> ⚠️ Warning: This code has the issue of the selection being broken

Oh, I forgot to sync this PR with our latest changes during the week. It should be fixed now! Thank you, once again!

**Comment by ThomazFB** [comment]
@atorresveiga Event validated and registered through https://github.com/Automattic/tracks-events-registration/pull/1741.

I'll move forward with merging this PR. Thanks!

**Pull request merged**